extensions/adapters/raster/src/main/java/org/locationtech/geowave/adapter/raster/operations/DeletePyramidLevelCommand.java

```diff
@@ -0,0 +1,222 @@
+        statsStore.getDataStatistics(
+            adapterIdStore.getAdapterId(adapter.getTypeName()),
+            OverviewStatistics.STATS_TYPE)) {
+      while (it.hasNext()) {
+        final InternalDataStatistics<?, ?, ?> next = it.next();
+        if (next instanceof OverviewStatistics) {
+          ovStats = (OverviewStatistics) next;
+          break;
+        }
+      }
+    }
+    if (ovStats == null) {
+      LOGGER.error("Unable to find overview stats for coverage " + adapter.getTypeName());
+      return;
+    }
+    if (!ovStats.removeResolution(res)) {
+      LOGGER.error("Unable to remove resolution for pyramid level " + level);
+      return;
+    }
+    try (CloseableIterator<InternalDataStatistics<?, ?, ?>> it =
+        statsStore.getDataStatistics(
+            adapterIdStore.getAdapterId(adapter.getTypeName()),
+            i.getName(),
+            PartitionStatistics.STATS_TYPE)) {
+      while (it.hasNext()) {
+        final InternalDataStatistics<?, ?, ?> next = it.next();
+        if (next instanceof PartitionStatistics) {
+          pStats = (PartitionStatistics) next;
+          break;
+        }
+      }
+    }
+    final byte[][] predefinedSplits = i.getIndexStrategy().getPredefinedSplits();
+    // this should account for hash partitioning if used
+    final List<ByteArray> partitions = new ArrayList<>();
+    if ((predefinedSplits != null) && (predefinedSplits.length > 0)) {
+      for (final byte[] split : predefinedSplits) {
+        partitions.add(new ByteArray(Arrays.append(split, level.byteValue())));
+      }
+    } else {
+      partitions.add(new ByteArray(new byte[] {level.byteValue()}));
+    }
+    if (pStats == null) {
+      LOGGER.error(
+          "Unable to find partition stats for coverage "
+              + adapter.getTypeName()
+              + " and index "
+              + i.getName());
+      return;
+    }
+    for (final ByteArray p : partitions) {
+      if (!pStats.getPartitionKeys().remove(p)) {
+        LOGGER.error(
+            "Unable to remove partition " + p.getHexString() + " for pyramid level " + level);
+        return;
+      }
+    }
+    statsStore.setStatistics(ovStats);
+    statsStore.setStatistics(pStats);
+
+    for (final ByteArray p : partitions) {
+      store.delete(
+          QueryBuilder.newBuilder().constraints(
+              QueryBuilder.newBuilder().constraintsFactory().prefix(
+                  p.getBytes(),
+                  null)).addTypeName(adapter.getTypeName()).indexName(i.getName()).build());
+    }
+  }
+
+  private static boolean isRaster(final DataTypeAdapter<?> adapter) {
+    if (adapter instanceof InternalDataAdapter) {
+      return isRaster(((InternalDataAdapter) adapter).getAdapter());
+    }
+    return adapter instanceof RasterDataAdapter;
+  }
+
+  public List<String> getParameters() {
+    return parameters;
+  }
+
+  public void setParameters(final String inputStore, final String outputStore) {
+    parameters = new ArrayList<>();
```

Comment:
This command only expects a single parameter.

Reply:
good catch